uqcsbot/starboard.py

```diff
@@ -40,20 +61,108 @@ def _rm_big_ratelimit(self, id: int):
         """ Callback to remove a message from the big-ratelimited list """
         self.big_blocked_messages.remove(id)
     
-    def _query_sb_message(self, recv: int) -> Optional[int]:
-        """ Get the starboard message ID corresponding to the recieved message """
+    @app_commands.default_permissions(manage_messages=True)
+    async def context_blacklist_sb_message(self, interaction: discord.Interaction, message: discord.Message):
+        """ Blacklists a message from being starboarded. If the message is already starboarded, also deletes it. """
         db_session = self.bot.create_db_session()
-        id = db_session.query(models.Starboard).filter(models.Starboard.recv == recv).one_or_none()
+        entry = db_session.query(models.Starboard).filter(models.Starboard.recv == message.id)
+        query_val = entry.one_or_none()
+
+        if query_val is not None:
+            if query_val.sent is None:
+                return
+
+            await (await self.starboard_channel.fetch_message(query_val.sent)).delete()
+            query_val.sent = None
+        else:
+            db_session.add(models.Starboard(recv=message.id, recv_location=message.channel.id, sent=None))
+
+        db_session.commit()
         db_session.close()
 
-        if id is not None:
-            return id.sent
-        return id
+        await interaction.response.send_message(f"Blacklisted message {message.id}.")
```

Comment:
Just a comment, is this worth logging in the admin-alerts channel as well?

Reply:
yes